File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/comum/validators/Min.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  org.hibernate.validator.ValidatorClass
 */
package br.jus.trt8.pjecalc.negocio.comum.validators;

import br.jus.trt8.pjecalc.negocio.comum.validators.MinValidator;
import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.hibernate.validator.ValidatorClass;

@Target(value={ElementType.FIELD})
@Retention(value=RetentionPolicy.RUNTIME)
@ValidatorClass(value=MinValidator.class)
@Documented
public @interface Min {
    public String value();

    public byte[] groups() default {};

    public String condition() default "";

    public String message() default "{MSG0004}";
}

File: src/lib/pjecalc/core/comum/validators/min.ts
/**
 * PJe-Calc v2.15.1 — Min (annotation TS-adaptada)
 * Porte de: br.jus.trt8.pjecalc.negocio.comum.validators.Min
 */
export interface MinSpec {
  value: string; // valor mínimo em string (BigDecimal em Java)
  groups?: number[];
  condition?: string;
  message?: string; // default "{MSG0004}"
}

export interface Min_ {
  value: string;
  groups: number[];
  condition: string;
  message: string;
}

export function min(spec: MinSpec): Min_ {
  return {
    value: spec.value,
    groups: spec.groups ?? [],
    condition: spec.condition ?? '',
    message: spec.message ?? '{MSG0004}',
  };
}
